tools/rewriter/GenCallAsm.cpp

```diff
@@ -1001,7 +1025,7 @@ std::string emit_asm_wrapper(AbiSignature &sig,
   add_asm_line(aw, ".type "s + wrapper_name + ", @function");
   add_asm_line(aw, wrapper_name + ":");
 
-  emit_prologue(aw, caller_pkey, target_pkey, arch);
+  emit_prologue(aw, caller_pkey, target_pkey, arch, target_post_condition_name.has_value());
```

Comment:
nit: maybe add a brief comment on why the prologue depends on whether we're doing post-condition validation or not

Reply:
and that the `pushq`s are undone when emitting the post-validation call (without any context I would assume the epilogue undoes everything done in the prologue).

Reply:
Fixed in 1dcaf11.